orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -85,12 +85,24 @@ class DependentPipelineStarter implements ApplicationContextAware {
       it.expectedArtifactIds ?: []
     }
 
-    // we are following a similar approach as triggers above
+    // following a similar approach as triggers above
     // expectedArtifacts can be used in triggers and stages
-    // for now we identified DeployManifestStage
-    // in ResolveDeploySourceManifestTask using ManifestEvaluator.getRequiredArtifacts
-    def requiredArtifactIds = pipelineConfig.get("stages", []).collectMany {
-      it.requiredArtifactIds ?: []
+    def requiredArtifactIds = pipelineConfig.get("stages", []).collectMany { stage ->
+      def artifactIds = []
+
+      //DeployManifestStage in ResolveDeploySourceManifestTask using ManifestEvaluator.getManifestArtifact
+      artifactIds += stage.manifestArtifactId ?: []
+
+      // DeployManifestStage in ResolveDeploySourceManifestTask using ManifestEvaluator.getRequiredArtifacts
+      artifactIds += stage.requiredArtifactIds ?: []
+
+      // BakeManifestStage in CreateBakeManifestTask using BakeManifestContext for Kustomize
+      artifactIds += stage.inputArtifact?.id ?: []
+
+      // BakeManifestStage in CreateBakeManifestTask using BakeManifestContext for Helm
```

Comment:
Are these comments accurate? What I mean is I understand that `Helm` bakes with `inputArtifacts` but other stages like `BakeCloudFoundry` also utilize this?

Can we make this more generic and say something more like

```
// Some bake manifest stages may populate the inputArtifacts array. So we need to look to for those ids as well
```

And we may be able to get rid of the other comments as they seem to be very specific but don't capture the whole picture

Reply:
Maybe even explain why we are looking for these ids as opposed to the expected artifact. Would be good to just document here saying we dont know how consuming pipelines will utilize artifacts from the parent pipeline, so we rely on any input artifact as opposed to the expected artifact